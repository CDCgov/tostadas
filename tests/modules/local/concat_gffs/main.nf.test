nextflow_process {

    name "Test Process CONCAT_GFFS"
    script "modules/local/concat_gffs/main.nf"
    process "CONCAT_GFFS"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                input[0] = file("${projectDir}/assets/ref/ref.MPXV.NC063383.v7.gff")
                input[1] = [ [ batch_id: "batch_1",  batch_tsv: file("${projectDir}/tests/modules/local/prep_submission/batch_1.tsv"), sample_id: "NY0006" ],
                               file("${projectDir}/assets/sample_fastas/mpox/NY0006.fasta"),
                               file("${projectDir}/tests/modules/local/concat_gffs/NY0006.fasta.out.gff"),
                               file("${projectDir}/tests/modules/local/concat_gffs/NY0006.liftoff-orig.gff")]
                """
            }
        }

        then {
            assert process.success
        }

    }

}
