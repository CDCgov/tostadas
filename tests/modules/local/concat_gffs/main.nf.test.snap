{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "batch_id": "batch_1",
                            "batch_tsv": "batch_1.tsv:md5,4ab00a8f04e89e82dd5504f5ca0c0a6c",
                            "sample_id": "NY0006"
                        },
                        "NY0006_reformatted.gff:md5,0b73232ae0199cad448f157d5fc6559f"
                    ]
                ],
                "1": [
                    [
                        {
                            "batch_id": "batch_1",
                            "batch_tsv": "batch_1.tsv:md5,4ab00a8f04e89e82dd5504f5ca0c0a6c",
                            "sample_id": "NY0006"
                        },
                        "NY0006_annotation_error.txt:md5,ba7064ab567293dd6a52ae240bf9d77f"
                    ]
                ],
                "2": [
                    [
                        {
                            "batch_id": "batch_1",
                            "batch_tsv": "batch_1.tsv:md5,4ab00a8f04e89e82dd5504f5ca0c0a6c",
                            "sample_id": "NY0006"
                        },
                        "NY0006.tbl:md5,228ca0a298dfa042f7d54cc85739c19d"
                    ]
                ],
                "errors": [
                    [
                        {
                            "batch_id": "batch_1",
                            "batch_tsv": "batch_1.tsv:md5,4ab00a8f04e89e82dd5504f5ca0c0a6c",
                            "sample_id": "NY0006"
                        },
                        "NY0006_annotation_error.txt:md5,ba7064ab567293dd6a52ae240bf9d77f"
                    ]
                ],
                "gff": [
                    [
                        {
                            "batch_id": "batch_1",
                            "batch_tsv": "batch_1.tsv:md5,4ab00a8f04e89e82dd5504f5ca0c0a6c",
                            "sample_id": "NY0006"
                        },
                        "NY0006_reformatted.gff:md5,0b73232ae0199cad448f157d5fc6559f"
                    ]
                ],
                "tbl": [
                    [
                        {
                            "batch_id": "batch_1",
                            "batch_tsv": "batch_1.tsv:md5,4ab00a8f04e89e82dd5504f5ca0c0a6c",
                            "sample_id": "NY0006"
                        },
                        "NY0006.tbl:md5,228ca0a298dfa042f7d54cc85739c19d"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-28T16:21:27.476767514"
    }
}